src/pages/[...slug].tsx

```diff
@@ -1,110 +1,75 @@
-import {
-  GetStaticPathsContext,
-  GetStaticPathsResult,
-  GetStaticPropsContext,
-  GetStaticPropsResult,
-} from 'next'
-import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
-import { drupalClient } from '@/utils/drupalClient'
-import { entityMeta } from '@/lib/delegators/entityMetaProvider'
-import { NodeTypes } from '@/types/node'
-import { getGlobalElements } from '@/lib/context/getGlobalElements'
-import Layout from '@/components/layout'
-import { generalEntityDataService } from '@/lib/delegators/generalEntityDataService'
+import * as React from "react"
+import { GetStaticPathsResult } from "next"
 
-interface EntityProps {
-  entityProps: any
+import { drupalClient } from "@/utils/drupalClient"
+import { queries } from "@/data/queries"
+import { NewsStoryPageProps } from "../types"
+import Layout from "@/components/layout"
+import { NewsStoryFull } from "@/components/news_story"
+
+const RESOURCE_TYPES = ["node--news_story"] as const
+
+interface ResourcePageProps {
+  resource: NewsStoryPageProps
 }
 
-/** Return the layout and primary component and props. */
-export default function Page({ entityProps, props }) {
-  const Component = entityMeta[entityProps.type].component
+export default function ResourcePage({ resource }: ResourcePageProps) {
+  if (!resource) return null
 
   return (
-    <Layout props={props}>
-      <Component {...entityProps} />
+    <Layout>
+      {resource.type === "node--news_story" && <NewsStoryFull {...resource} />}
```

Comment:
I have not yet created an equivalent of our `<Node />` component that does delegation, but we will want something similar to that which maps a resource type to a component for page rendering.